lib/encoder/include/Encoder.h

```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+namespace Encoder {
+class Encoder {
+ public:
+  virtual ~Encoder(){};
+
+  virtual bool getAngleDeg(float &angle)                 = 0;
+  virtual bool getAngularVelocityDegPerSec(float &speed) = 0;
```

Comment:
Some encoders only give position readings and not velocity, are we ok with having this function optional?

Reply:
i would say that all encoders should have a get_angle and get_speed function. the get_speed can just differentiate last angle to current angle wrt to time. basically just have a timer that times delta t